tests/gtest.cpp

```diff
@@ -0,0 +1,485 @@
+#include <gtest/gtest.h>
+#include "../tests/scheduler_test_system.hpp"
+#include "scheduler_functions.hpp"
+class SchedulerTest : public ::testing::Test
+{
+protected:
+    DeploymentSchedule_t deploymentSchedule[4];
+    DeploymentSchedule_t* nextEvent;
+    uint32_t nextEventTime;
+    SchedulerTest()
+    {
+        deploymentSchedule[0] = { SS_ensembleAFunc, SS_ensembleAInit, 1,
+                millis(), 2000, UINT32_MAX, 0, 0, 0, nullptr, 400, (char)65 };
+        deploymentSchedule[1] = { SS_ensembleBFunc, SS_ensembleBInit, 1,
+                millis(), 2000, UINT32_MAX, 0, 0, 0, nullptr, 200, (char)66 };
+        deploymentSchedule[2] = { SS_ensembleCFunc, SS_ensembleCInit, 1,
+                millis(), 2000, UINT32_MAX, 0, 0, 0, nullptr, 600, (char)67 };
+        deploymentSchedule[3] = { nullptr, nullptr, 0, 0, 0, 0, 0, 0, 0,
+                nullptr };
+    }
+    void schedule2()
+    {
+        deploymentSchedule[2] = { nullptr, nullptr, 0, 0, 0, 0, 0, 0, 0,
+            nullptr };
+    }
+    void schedule3()
+    {
+        deploymentSchedule[0] = { SS_ensembleAFunc, SS_ensembleAInit, 1,
+                millis(), 500, UINT32_MAX, 0, 0, 0, nullptr,200,(char)65 };
+        deploymentSchedule[1] = { SS_ensembleBFunc, SS_ensembleBInit, 1,
+                millis(), 200, UINT32_MAX, 0, 0, 0, nullptr,100,(char)66 };
+        deploymentSchedule[2] = { SS_ensembleCFunc, SS_ensembleCInit, 1,
+                millis(), 800, UINT32_MAX, 0, 0, 0, nullptr,250,(char)67 };
+    }
+    void SetUp() override
+    {
+        nextEvent = nullptr;
+        nextEventTime = 0;
+        setTime(0);
+        SCH_initializeSchedule(deploymentSchedule, millis());
+
+    }
+
+    void TearDown() override
+    {
+    }
+    void verifyNextEvent(char expectedTaskName, uint32_t expectedStart,
+        uint32_t expectedEnd, uint32_t additionalTime)
+    {
+        ASSERT_EQ(expectedTaskName, nextEvent->taskName);
+        setTime(nextEventTime);
+        ASSERT_EQ(expectedStart, nextEventTime);
+        addTime(nextEvent->maxDuration);
+        addTime(additionalTime);
+        ASSERT_EQ(expectedEnd, millis());
+    }
+    void verifyNextEvent(char expectedTaskName, uint32_t expectedStart,
+        uint32_t expectedEnd)
+    {
+        ASSERT_EQ(expectedTaskName, nextEvent->taskName);
+        setTime(nextEventTime);
+        ASSERT_EQ(expectedStart, nextEventTime);
+        addTime(nextEvent->maxDuration);
+        ASSERT_EQ(expectedEnd, millis());
+    }
```

Comment:
What do these functions do?  Are they simply a verification function, or do they actually change the state of the simulation?

Reply:
They do change the state of the simulation. They're just meant to clean up the tests for reading but I can put it back in the tests if preferred.

Reply:
In this case, `verify` might not be the right name - maybe a different name would be appropriate